src/openrct2/management/NewsItem.cpp

```diff
@@ -207,73 +207,70 @@ static int32_t news_item_get_new_history_slot()
  *
  *  rct2: 0x0066BA74
  */
-void news_item_get_subject_location(int32_t type, int32_t subject, int32_t* x, int32_t* y, int32_t* z)
+std::optional<CoordsXYZ> news_item_get_subject_location(int32_t type, int32_t subject)
 {
-    Ride* ride;
-    Peep* peep;
-    Vehicle* vehicle;
+    std::optional<CoordsXYZ> subjectLoc{ std::nullopt };
 
     switch (type)
     {
         case NEWS_ITEM_RIDE:
-            ride = get_ride(subject);
+        {
+            Ride* ride = get_ride(subject);
             if (ride == nullptr || ride->overall_view.isNull())
             {
-                *x = LOCATION_NULL;
                 break;
             }
-            *x = ride->overall_view.x + 16;
-            *y = ride->overall_view.y + 16;
-            *z = tile_element_height({ *x, *y });
+            auto rideViewCentre = ride->overall_view.ToTileCentre();
+            subjectLoc = CoordsXYZ{ rideViewCentre, tile_element_height(rideViewCentre) };
             break;
+        }
         case NEWS_ITEM_PEEP_ON_RIDE:
-            peep = GET_PEEP(subject);
-            *x = peep->x;
-            *y = peep->y;
-            *z = peep->z;
-            if (*x != LOCATION_NULL)
+        {
+            Peep* peep = GET_PEEP(subject);
+            subjectLoc = CoordsXYZ{ peep->x, peep->y, peep->z };
+            if (subjectLoc->x != LOCATION_NULL)
                 break;
 
             if (peep->state != 3 && peep->state != 7)
```

Comment:
Do we have constants for these? If so, please use them.

Reply:
I don't know, I can update them on another PR, as I didn't touch these lines, so I think it'd be weird.

Reply:
I'd count it as "Boy Scout principle", you're modifying the function anyway.

I'm ok with doing them in another PR, though.